src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsEndpoint.java

```diff
@@ -6,21 +6,42 @@
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.StreamingOutput;
+import javax.ws.rs.core.Response.Status;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+
+import javax.servlet.http.HttpServletRequest;
 
 public class MetricsEndpoint implements RealmResourceProvider {
 
     // The ID of the provider is also used as the name of the endpoint
     public final static String ID = "metrics";
 
+    private static final boolean DISABLE_OPENSHIFT_EXTERNAL_ACCESS = Boolean.parseBoolean(System.getenv("DISABLE_OPENSHIFT_EXTERNAL_ACCESS"));
+
     @Override
     public Object getResource() {
         return this;
     }
 
     @GET
     @Produces(MediaType.TEXT_PLAIN)
-    public Response get() {
+    public Response get(@Context HttpServletRequest servletRequest, @Context HttpHeaders headers) {
+
+        if (DISABLE_OPENSHIFT_EXTERNAL_ACCESS){
+            if ( !headers.getRequestHeader("x-forwarded-host").isEmpty() ) {
```

Comment:
@CathalOConnorRH Can we rename the flag to `DISABLE_EXTERNAL_ACCESS`? The spi itself, nor the `x-forwarded-host` are not OpenShift specific. 

The value of the header is not checked. So any requester who sets the header would be permitted. Is that ok?

Reply:
updated env var and readme